scripts/migration/migrate_menas.py

```diff
@@ -0,0 +1,433 @@
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        prog='migrate_menas',
+        description='Menas MongoDB migration script.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter  # prints default values, too, on help (-h)
+    )
+
+    parser.add_argument('-n', '--dryrun', action='store_true', default=defaults.dryrun,
+                        help="if specified, skip the actual synchronization, just print what would be copied over.")
+    parser.add_argument('-v', '--verbose', action="store_true", default=defaults.verbose,
+                        help="prints extra information while running.")
+
+    parser.add_argument('source', metavar="SOURCE",
+                        help="connection string for source MongoDB")
+    parser.add_argument('target', metavar="TARGET",
+                        help="connection string for target MongoDB")
+
+    parser.add_argument('-t', '--target-database', dest="targetdb", default=defaults.target_db_name,
+                        help="Name of db on target to migrate to.")
+
+    input_options_group = parser.add_mutually_exclusive_group(required=True)
+    input_options_group.add_argument('-d', '--datasets', dest='datasets', metavar="DATASET_NAME", default=[],
+                                     nargs="+", help='list datasets to migrate')
+    input_options_group.add_argument('-m', '--mapping-tables', dest="mtables", metavar="MTABLE_NAME", default=[],
+                                     nargs="+", help='list datasets to migrate')
+
+    return parser.parse_args()
+
+
+def run(parsed_args: argparse.Namespace):
+    source = parsed_args.source
+    target = parsed_args.target
+    target_db_name = parsed_args.targetdb
+
+    print('Menas mongo migration')
+    print('Running with settings: dryrun={}, verbose={}'.format(dryrun, verbose))
+    print("Using migration #: '{}'".format(migration_hash))
+    print('  source connection-string: {}'.format(source))
+    print('  source DB: {}'.format(SOURCE_DB_NAME))
+    print('  target connection-string: {}'.format(target))
+    print('  target DB: {}'.format(target_db_name))
+
+    dataset_names = parsed_args.datasets
+    mt_names = parsed_args.mtables
+    if dataset_names:
+        print('dataset names supplied: {}'.format(dataset_names))
+        migrate_collections_by_ds_names(source, target, target_db_name, dataset_names)
+    elif mt_names:
+        print('mapping table names supplied: {}'.format(mt_names))
+        migrate_collections_by_mt_names(source, target, target_db_name, mt_names)
+    else:
+        # should not happen (-d/-m is exclusive and required)
+        raise Exception("Invalid run options: DS names (-d ds1 ds2 ...).. or MT names (-m mt1 mt2 ... ) must be given.")
+
+    print("Done.")
+
+
+if __name__ == '__main__':
+    args = parse_args()
+
+    # script-global flags
+    dryrun = args.dryrun
+    verbose = args.verbose
```

Comment:
Are these like global values?

Reply:
yes, they are. That is why the methods use it even thought their signature does not contain it.

Reply:
I have checked sources like https://www.python.org/dev/peps/pep-0008/#global-variable-names and https://pythonguides.com/python-naming-conventions/ None of them suggest a prefix.
There is a `global` keyword, but it is for write access to global vars, which we don't need https://www.programiz.com/python-programming/global-keyword

Nevertheless, I have at least put them all in the same place (`__main__`) which should make it pretty readable.

Also, to minimize exposure, I have made the `dryrun` non-global`, as it does not make a lot of mess with bloating of the signatures (as opposed to `verbose` might). Also, it is fair for `dryrun` to be in the method signatures as it has significant effect on the script run. Verbose only results in more prints, so no significant behavior change.